packages-exp/functions-exp/src/index.node.ts

```diff
@@ -16,10 +16,22 @@
  */
 import { registerVersion } from '@firebase/app-exp';
 import { registerFunctions } from './config';
-import 'isomorphic-fetch';
+import 'whatwg-fetch';
+import nodeFetch from 'node-fetch';
 
 import { name, version } from '../package.json';
 
+/**
+ * Patch global object with node-fetch. whatwg-fetch polyfill patches other global
+ * fetch objects such as Headers. Then we override the implemenation of `fetch()`
+ * itself with node-fetch.
+ * https://github.com/node-fetch/node-fetch#loading-and-configuring-the-module
+ * node-fetch type deviates somewhat from fetch spec:
+ * https://github.com/apollographql/apollo-link/issues/513#issuecomment-548219023
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+(global as any).fetch = (nodeFetch as unknown) as typeof fetch;
```

Comment:
I'm worried about introducing hard to debug issues in developers' app by patching globals. For example, they might use a different fetch polyfill which can be overwritten by our `node-fetch` polyfill depending on the import order. node-fetch might have a bug that the other polyfill doesn't have. It will be really hard for developers to track down the issue.

Though the chance of this kind of bug is slim, but it's not impossible, I think it's a good practice to not patch globals to future proof the SDK and developers' apps.

Reply:
Fixed, it is passed in now.